AWSSDKSwiftCLI/Sources/SPR/UploadManifest.swift

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSS3
+import Smithy
+import SmithyStreams
+import AWSCLIUtils
+
+extension SPRPublisher {
+
+    func uploadManifest() async throws {
```

Comment:
This function:
- Gets the package manifest (i.e. `Package.swift` for the package being published) from S3 to verify that it does not already exist, then
- Puts the package manifest to S3.